Microsoft.Toolkit.Uwp.UI/Extensions/Common/StringExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Toolkit.Uwp.UI.Extensions.Common
+{
+    /// <summary>
+    /// All common string extensions should go here
+    /// </summary>
+    public static class StringExtensions
+    {
+        /// <summary>
+        /// Returns whether said string is a valid email or not.
+        /// Uses general Email Regex (RFC 5322 Official Standard) from emailregex.com
+        /// </summary>
+        /// <returns><c>true</c> for valid email.<c>false</c> otherwise</returns>
+        public static bool IsEmail(this string str)
+        {
+            return Regex.IsMatch(str, "(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)\\])");
```

Comment:
To make the TextBoxRegEx change easier, you could pull out the regex pattern string as an `internal` constant and just have the TextBoxRegEx switch statement reference that.

@nmetulev think it'd actually be useful to pull all the RegEx expressions out of the TextBoxRegEx class and put them here?  Then we can have nice helper methods for string for each of the different types?

Reply:
I can do another PR for moving everything related to `TextBoxRegEx` to this helper method. For now I will update this PR moving the string constants separate.

Reply:
Makes sense. You should still move the extension to the Microsoft.Toolkit.Uwp package in this PR